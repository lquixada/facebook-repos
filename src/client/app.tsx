import React from 'react'
import {AppContainer} from 'react-hot-loader'
import {Provider} from 'react-redux'
import {renderRoutes} from 'react-router-config'
import {BrowserRouter} from 'react-router-dom'

import routes from '../shared/routes'
import store from './store'

const App: any = AppContainer

export default () => (
  <App>
    <Provider store={store}>
      <BrowserRouter>
        {renderRoutes(routes)}
      </BrowserRouter>
    </Provider>
  </App>
)
