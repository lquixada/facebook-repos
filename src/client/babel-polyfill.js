// Issue and solution: https://github.com/babel/babel-loader/issues/401#issuecomment-365650944
if (!global._babelPolyfill) {
  require('babel-polyfill')
}
