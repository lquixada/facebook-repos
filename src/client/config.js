// Client-side config
const reposHost = process.env.NODE_ENV === 'production' ? 'https://repos.lquixada.com' : 'http://localhost:3001'

export default {
  staticUrl: process.env.NODE_ENV === 'production' ? 'https://static.lquixada.com/repos' : '/assets',
  github: {
    contributorsUrl: `${reposHost}/api/:repo/contributors`,
    repoUrl: `${reposHost}/api/:repo`,
    reposUrl: `${reposHost}/api/repos`,
    webUrl: 'https://github.com'
  }
}
