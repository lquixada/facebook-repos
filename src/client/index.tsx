import React from 'react'
import ReactDOM from 'react-dom'
import {AppContainer} from 'react-hot-loader'
import {Provider} from 'react-redux'
import {renderRoutes} from 'react-router-config'
import {BrowserRouter} from 'react-router-dom'
import 'regenerator-runtime/runtime'

import store from './store'

const App: any = AppContainer
const renderApp = () => {
  const routes = require('../shared/routes').default

  ReactDOM.hydrate(
    <App>
      <Provider store={store}>
        <BrowserRouter>
          {renderRoutes(routes)}
        </BrowserRouter>
      </Provider>
    </App>
    , document.getElementById('app'),
  )
}

renderApp()

if (module.hot) {
  module.hot.accept('../shared/routes', renderApp)
}
