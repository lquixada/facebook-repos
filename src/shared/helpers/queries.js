import gql from 'graphql-tag'

import {getClient} from './client'

export const fetchContributors = async ({owner, repoName, page}) => {
  const client = getClient()
  const {data} = await client.query({
    query: gql`
      {
        contributors(owner: "${owner}", repo: "${repoName}", page: ${page}) {
          nextPage
          data {
            login
            html_url
            avatar_url
          }
        }
      }
    `
  })

  return data
}

export const fetchRepo = async ({ owner, repoName }) => {
  const client = getClient()
  const {data} = await client.query({
    query: gql`
      {
        repo(owner: "${owner}", name: "${repoName}") {
          name
          description
          html_url
          stargazers_count
          forks_count
          subscribers_count
          open_issues_count
          contributors {
            nextPage
            data {
              login
              html_url
              avatar_url
            }
          }
          license {
            name
          }
        }
      }
    `
  })

  return data
}

export const fetchCounts = async ({ owner }) => {
  const client = getClient()
  const {data} = await client.query({
    query: gql`
      {
        repoCount(owner: "${owner}") {
          name
          count
        }
      }
    `
  })

  return data
}

export const fetchAll = async ({ owner, repoName }) => {
  const client = getClient()
  const {data} = await client.query({
    query: gql`
      {
        repoCount(owner: "${owner}") {
          name
          count
        }
      
        repo(owner: "${owner}", name: "${repoName}") {
          name
          description
          html_url
          stargazers_count
          forks_count
          subscribers_count
          open_issues_count
          contributors {
            nextPage
            data {
              login
              html_url
              avatar_url
            }
          }
          license {
            name
          }
        }
      }
    `
  })

  return data
}
