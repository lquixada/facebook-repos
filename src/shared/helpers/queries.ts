import gql from 'graphql-tag'

import {getClient} from './client'

export const fetchContributors = async (variables: object) => {
  const client = getClient()
  const {data} = await client.query({
    query: gql`
      query($owner: String!, $repoName: String!, $page: Int) {
        contributors(owner: $owner, repo: $repoName, page: $page) {
          nextPage
          data {
            login
            html_url
            avatar_url
          }
        }
      }
    `,
    variables,
  })

  return data
}

export const fetchRepo = async (variables: object) => {
  const client = getClient()
  const {data} = await client.query({
    query: gql`
      query($owner: String!, $repoName: String!) {
        repo(owner: $owner, name: $repoName) {
          name
          description
          html_url
          stargazers_count
          forks_count
          subscribers_count
          open_issues_count
          contributors {
            nextPage
            data {
              login
              html_url
              avatar_url
            }
          }
          license {
            name
          }
        }
      }
    `,
    variables,
  })

  return data
}

export const fetchCounts = async (variables: object) => {
  const client = getClient()
  const {data} = await client.query({
    query: gql`
      query($owner: String!) {
        counts(owner: $owner) {
          name
          count
        }
      }
    `,
    variables,
  })

  return data
}

export const fetchAll = async (variables: object) => {
  const client = getClient()
  const {data} = await client.query({
    query: gql`
      query($owner: String!, $repoName: String!) {
        counts(owner: $owner) {
          name
          count
        }

        repo(owner: $owner, name: $repoName) {
          name
          description
          html_url
          stargazers_count
          forks_count
          subscribers_count
          open_issues_count
          contributors {
            nextPage
            data {
              login
              html_url
              avatar_url
            }
          }
          license {
            name
          }
        }
      }
    `,
    variables,
  })

  return data
}
