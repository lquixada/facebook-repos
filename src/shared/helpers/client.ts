import ApolloClient from 'apollo-boost'
import config from '../../server/config'

const client = new ApolloClient({
  uri: config.graphql,
})

export const getClient = () => client
