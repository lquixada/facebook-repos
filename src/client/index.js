import 'babel-polyfill';
import React from 'react';
import ReactDOM from 'react-dom';
import {Provider} from 'react-redux';
import {BrowserRouter} from 'react-router-dom';
import {renderRoutes, matchRoutes} from 'react-router-config';
import {AppContainer} from 'react-hot-loader';
import {END} from 'redux-saga';

import {trigger} from '../shared/helpers';
import routes from '../shared/routes'; // eslint-disable-line no-unused-vars
import store from './store';

const renderApp = () => {
  const routes = require('../shared/routes').default;

  ReactDOM.hydrate(
    <AppContainer>
      <Provider store={store}>
        <BrowserRouter>
          {renderRoutes(routes)}
        </BrowserRouter>
      </Provider>
    </AppContainer>
    , document.getElementById('app')
  );
};

const matchs = matchRoutes(routes, window.location.pathname);

if (matchs.length === 0) {
  throw new Error(`React Router: Not found ${window.location.pathname}`);
}

if (!window.__INITIAL_STATE__) {
  store.runnedSagas.toPromise().then(renderApp);
  trigger('fetch', matchs, store.dispatch);
  store.dispatch(END);
} else {
  renderApp();
}

if (module.hot) {
  module.hot.accept('../shared/routes', renderApp);
}
