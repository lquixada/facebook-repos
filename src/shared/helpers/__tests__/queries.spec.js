import createMockClient from '../../../__tests__/mocked-client'
import {fetchContributors, fetchRepo, fetchRepos} from '../queries'

import * as helpers from '../client'

describe('Helpers (Queries)', () => {
  beforeEach(() => {
    const client = createMockClient()
    helpers.getClient = () => client
  })

  describe('fetchContributors', () => {
    it('returns contributors from a repo', async () => {
      const data = await fetchContributors('some-repo', 1)
      expect(data).toMatchSnapshot()
    })
  })

  describe('fetchRepo', () => {
    it('returns data from a repo', async () => {
      const data = await fetchRepo('some-repo')
      expect(data).toMatchSnapshot()
    })
  })

  describe('fetchRepos', () => {
    it('returns repos from an organization', async () => {
      const data = await fetchRepos()
      expect(data).toMatchSnapshot()
    })
  })
})
