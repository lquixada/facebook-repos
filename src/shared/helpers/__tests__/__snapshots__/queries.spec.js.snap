// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helpers (Queries) fetchContributors returns contributors from a repo 1`] = `
Object {
  "contributors": Object {
    "data": Array [
      Object {
        "avatar_url": "http://avatar_url/",
        "html_url": "http://html_url/",
        "login": "user1",
        Symbol(id): "$ROOT_QUERY.contributors({\\"page\\":1,\\"repo\\":\\"some-repo\\"}).data.0",
      },
    ],
    "nextPage": 2,
    Symbol(id): "$ROOT_QUERY.contributors({\\"page\\":1,\\"repo\\":\\"some-repo\\"})",
  },
  Symbol(id): "ROOT_QUERY",
}
`;

exports[`Helpers (Queries) fetchRepo returns data from a repo 1`] = `
Object {
  "contributors": Object {
    "data": Array [
      Object {
        "avatar_url": "http://avatar_url/",
        "html_url": "http://html_url/",
        "login": "user1",
        Symbol(id): "$ROOT_QUERY.contributors({\\"page\\":1,\\"repo\\":\\"some-repo\\"}).data.0",
      },
    ],
    "nextPage": 2,
    Symbol(id): "$ROOT_QUERY.contributors({\\"page\\":1,\\"repo\\":\\"some-repo\\"})",
  },
  "repo": Object {
    "description": "",
    "forks_count": 1,
    "html_url": "",
    "license": Object {
      "name": "",
      Symbol(id): "$ROOT_QUERY.repo({\\"name\\":\\"some-repo\\"}).license",
    },
    "name": "repo1",
    "open_issues_count": 1,
    "stargazers_count": 1,
    "subscribers_count": 1,
    Symbol(id): "$ROOT_QUERY.repo({\\"name\\":\\"some-repo\\"})",
  },
  Symbol(id): "ROOT_QUERY",
}
`;

exports[`Helpers (Queries) fetchRepos returns repos from an organization 1`] = `
Object {
  "repoCount": Array [
    Object {
      "count": 2,
      "name": "repo1",
      Symbol(id): "ROOT_QUERY.repoCount.0",
    },
  ],
  Symbol(id): "ROOT_QUERY",
}
`;
